.pre-commit-config.yaml

```
@@ -28,10 +28,20 @@ repos:
       - id: requirements-txt-fixer
       - id: mixed-line-ending
         args: ["--fix=auto"]
-  - repo: https://github.com/google/yapf
-    rev: v0.43.0
+  - repo: https://github.com/psf/black
+    rev: 23.11.0
     hooks:
-      - id: yapf
+    - id: black
+      language_version: python3
+  - repo: https://github.com/pycqa/isort
+    rev: 5.12.0
+    hooks:
+    - id: isort
+      args: ["--profile", "black", "--filter-files"]
+  - repo: https://github.com/PyCQA/flake8
+    rev: 6.1.0
+    hooks:
+    - id: flake8
```

If you want to allow slight overflows above 79 characters in comments or long lines, easing the line length restriction in flake8 slightly will help allow it.

```suggestion
    - id: flake8
      args: ["--max-line-length=88"]
```